builtin/common/serialize.lua

```diff
@@ -133,11 +133,13 @@ function core.serialize(x)
 			local vals = {}
 			local idx_dumped = {}
 			local np = nest_points[x]
-			for i, v in ipairs(x) do
-				if not np or not np[i] then
-					vals[#vals + 1] = dump_or_ref_val(v)
+			if vector.is_vector(x) ~= 2 then
```

Comment:
If this code is required for serialization not to break, that could be another backwards-compatibility problem.

Reply:
Modified vector tables that had numerical indexes (starting with 1), additional to x, y and z, won't work anymore anyways. Hence, iteration through the table with `ipairs` makes no sense here.
Things like `local v = vector.new() v[1] = "bla"` can't be preserved, `v[1]` now accesses `x`, which is intended.

Reply:
I know that ipairs on a vector makes no sense.
The problem here however is if someone has copy-paste serialization code, that code will now serialize vectors incorrectly.

Reply:
Such mods (if existent) will have to be fixed then.